### ISSUE TITLE
Link compilation failures back to original source location

### ISSUE THREAD

**Issue by haines** [cerbos/cerbos#702]
If I have a policy that does not match the schema, the `cerbos compile` error is difficult to trace back to my policy file. because internally it coerces the YAML to JSON which loses line/column number information.

For example, compiling

```yaml
apiVersion: api.cerbos.dev/v1
derivedRoles:
  name: organization_assignments
  definition:
    - name: manufacturer
      parentRoles:
        - user
      condition:
        match:
          expr: hasIntersection(R.attr.organizationAssignments.manufacturer, P.attr.organizations)
```

produces

```
Load failures
policies/derived_roles/organization_assignments.yaml: failed to unmarshal JSON: proto: (line 1:51): unknown field "definition"
```

which is confusing to the user because a) they are providing YAML not JSON and b) the error is on line 4:3 not line 1:51.

**Comment by haines** [comment]
If we can do this, it'd be great to then alter the output format to be more like other compilers. `pretty` might give a Clang-style format, and `plain` something like https://www.gnu.org/prep/standards/html_node/Errors.html (some editors can parse this out of the box).

**Comment by charithe** [comment]
This is something the definitely needs improvement. At one point I was investigating [CUE](https://cuelang.org) to do this. Unfortunately it didn't have support for disjoint unions (`oneOf`) at the time. Maybe that has changed now. I haven't checked recently.
 
I can't think of another way to do this without actually writing our own parser. In the long term that's probably what we'll have to do.

The output format is definitely meant to be parseable by editors (Makefile output). We don't have line numbers yet so it doesn't look like that, unfortunately :-)

**Comment by charithe** [comment]
Fixed in #1960 and #1968
